backend/src/main/java/ch/puzzle/pcts/repository/CertificateTypeRepository.java

```diff
@@ -8,9 +8,39 @@
 
 @Repository
 public interface CertificateTypeRepository extends SoftDeleteRepository<CertificateType, Long> {
-    List<CertificateType> findByCertificateKindNotAndDeletedAtIsNull(CertificateKind certificatekind);
+    List<CertificateType> findAllByCertificateKindAndDeletedAtIsNull(CertificateKind kind);
 
-    List<CertificateType> findByCertificateKindAndDeletedAtIsNull(CertificateKind certificatekind);
+    List<CertificateType> findAllByCertificateKindNotAndDeletedAtIsNull(CertificateKind kind);
 
-    Optional<CertificateType> findByName(String name);
-}
+    Optional<CertificateType> findByIdAndCertificateKindAndDeletedAtIsNull(Long id, CertificateKind kind);
+
+    Optional<CertificateType> findByIdAndCertificateKindNotAndDeletedAtIsNull(Long id, CertificateKind kind);
+
+    Optional<CertificateType> findByNameAndCertificateKindAndDeletedAtIsNull(String name, CertificateKind kind);
+
+    Optional<CertificateType> findByNameAndCertificateKindNotAndDeletedAtIsNull(String name, CertificateKind kind);
+
+    default List<CertificateType> findAllFromCertificateType() {
```

Comment:
```suggestion
    default List<CertificateType> findAllOfCertificateType() {
```
NIT: about the naming